File: Programmers/위장.py
def solution(clothes):
    answer = 0
    arr = {}
    
    for i in clothes:
        if not i[1] in arr:
            arr[i[1]] = [i[0]]
            arr[i[1]].append(0)
        else:
            arr[i[1]].append(i[0])
            
    for i in arr:
        if answer == 0:
            answer += len(arr[i])
        else:
            answer *= len(arr[i])
    
    return answer-1

File: Programmers/위장.js
function solution(clothes) {
	let answer = 0;
	let arr = {};

	for (let i = 0; i < clothes.length; i++) {
		if (arr[clothes[i][1]]) {
			arr[clothes[i][1]].push(clothes[i][0]);
		} else {
			arr[clothes[i][1]] = [clothes[i][0]];
			arr[clothes[i][1]].push(0);
		}
	}

	for (let i in arr) {
		if (answer === 0) {
			answer += arr[i].length;
		} else {
			answer *= arr[i].length;
		}
	}

	return answer - 1;
}
